src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -434,8 +439,32 @@ async def infer_runtime_platforms(request: RuntimePlatformsRequest) -> RuntimePl
         )
     except StopIteration:
         # Not a known runtime, so fallback to just passing a platform
-        # TODO: maybe this should be an error, and we require users to specify
-        # complete_platforms themselves?
+        version_modifier = "[inferred from interpreter constraints]" if inferred_from_ics else ""
+        version_adjective = "inferred" if inferred_from_ics else "specified"
+        known_runtimes_str = ", ".join(
+            FrozenOrderedSet(r.name for r in request.runtime.known_runtimes)
+        )
+        warn_or_error(
+            "2.26.0.dev0",
+            "implicitly resolving platforms for unknown FaaS runtimes",
+            softwrap(
+                f"""
+                Could not find a known runtime for the {version_adjective} Python version and machine architecture!
+
+                Python version: {version} {version_modifier}
+                Machine architecture: {request.architecture.value}
```

Comment:
I think the `softwrap` will cause this to be displayed all on one line:

```
Python version: 3.9 Machine architecture: x86_64
```

So I think we might need either: 

- make this into a list with `*` https://github.com/pantsbuild/pants/blob/fa583d6b36df4c7bb1c560dc5429d1bfb474ba51/src/python/pants/util/strutil.py#L254-L256
- make it display more nicely on one line, e.g.
  ```suggestion
                  Python version: {version} {version_modifier}; machine architecture: {request.architecture.value}
  ```

Either is fine by me!

Reply:
I went with the bulleted list format and rolled in the list of valid runtime values as part of the list as well!